backend/sozisel/lib/sozisel/model/sessions.ex

```diff
@@ -138,7 +137,13 @@ defmodule Sozisel.Model.Sessions do
       |> Map.from_struct()
       |> Map.merge(%{id: nil, user_id: user.id, agenda_entries: agenda_entries})
 
-    create_template_with_agenda(copy_template)
+    new_template = create_template_with_agenda(copy_template)
+
+    Enum.each(Events.list_template_events(template.id), fn event ->
```

Comment:
First of all, this should be done inside create_template_and_agenda or a similar function so all those operation stay inside a single database transaction. Second of all this formating is hard to look at, use pipeline operators as much as possible
```elixir
template.id
|> Events.list_template_events()
|> Enum.each(fn event ->
  event
  |> Map.from_struct()
  |> Map.merge(%{id: new_template.id})
  |> Events.create_event()
end)
```

And as said previously, I believe that this is wrong, you should not set id to `new_template.id` but rather `session_template_id` or whatever is right in `events` schema. And once more, this should be in a single transaction with creating the whole template with the agenda points when cloning.